app/Services/Avatars/LocalAvatarService.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\Services\Avatars;
+
+use Illuminate\Support\Facades\Cache;
+
+class LocalAvatarService
+{
+    public function generateSvgAvatar(
+        string $name,
+        string $backgroundColor = '111827',
+        string $textColor = 'FFFFFF',
+        int $size = 128
+    ): string {
+        $initials = $this->getInitials($name);
+        $fontSize = $size * 0.4;
+
+        $svg = <<<SVG
+<svg xmlns="http://www.w3.org/2000/svg" width="{$size}" height="{$size}" viewBox="0 0 {$size} {$size}">
+    <rect width="{$size}" height="{$size}" fill="#{$backgroundColor}"/>
+    <text x="50%" y="50%" dominant-baseline="central" text-anchor="middle" font-family="Arial, sans-serif" font-size="{$fontSize}" fill="#{$textColor}" font-weight="500">{$initials}</text>
+</svg>
+SVG;
+
+        return $svg;
+    }
+
+    protected function getInitials(string $name): string
+    {
+        $initials = str($name)
+            ->trim()
+            ->explode(' ')
+            ->map(fn (string $segment): string => filled($segment) ? mb_substr($segment, 0, 1) : '')
+            ->join('');
+
+        return strtoupper((string) $initials);
+    }
+
+    public function generateDataUri(
+        string $name,
+        string $backgroundColor = '111827',
+        string $textColor = 'FFFFFF',
+        int $size = 128
+    ): string {
+        $backgroundColor = ltrim($backgroundColor, '#');
+        $cacheKey = "avatar:{$name}:{$backgroundColor}:{$textColor}:{$size}";
+
+        return Cache::remember($cacheKey, now()->addDay(), function () use ($name, $backgroundColor, $textColor, $size) {
+            $svg = $this->generateSvgAvatar($name, $backgroundColor, $textColor, $size);
+
+            return 'data:image/svg+xml;base64,' . base64_encode($svg);
+        });
+    }
+
+    public function generateColorFromName(string $name): string
+    {
+        $hash = md5($name);
+        $hue = hexdec(substr($hash, 0, 2)) / 255 * 360;
+        $saturation = 60;
+        $lightness = 45;
+
+        return $this->hslToHex($hue, $saturation, $lightness);
+    }
+
+    protected function hslToHex(float $h, float $s, float $l): string
```

Comment:
```suggestion
    public function generateColorFromName(string $name): string
    {
        $hash = md5($name);
        $hue = hexdec(substr($hash, 0, 2)) / 255 * 360;

        return $this->hslToHex($hue);
    }

    protected function hslToHex(float $hue, ?float $saturation = 60, float $lightness = 45): string
```
Please use descriptive variable names.